reference/features/security.md

```diff
@@ -0,0 +1,237 @@
+  "createdAt": "2022-01-01T10:00:00Z",
+  "updatedAt": "2022-01-01T10:00:00Z"
+}
+```
+
+It is good practice to always set an expiry date when creating a new API key. If you are sure that is not necessary in your application, you can create an API key with no expiry date by explicitly passing a `null` value to `expiresAt`.
+
+### Listing API keys
+
+You can use the [list keys endpoint](/reference/api/keys.md) to obtain information on any active key in your MeiliSearch instance. This is useful when you need an overview of existing keys and their permissions.
+
+[`GET /keys`](/reference/api/keys.md#get-all-keys) returns a full list of all existing keys. **Expired keys will appear in the response, but deleted keys will not**. As with creating, deleting, and updating API keys, you need the master key to access this endpoint.
+
+[`GET /keys/:key`](/reference/api/keys.md#get-one-key) returns information on a single key. `:key` should be replaced with the full `key` value obtained during key creation.
+
+We can query our instance to confirm which active keys can search our `patient_medical_records` index:
+
+<CodeSamples id="security_guide_list_keys_1" />
+
+```json
+{
+  "results": [
+    {
+      "description": "Default Search API Key (Use it to search from the frontend code)",
+      "key": "0a6e572506c52ab0bd6195921575d23092b7f0c284ab4ac86d12346c33057f99",
+      "actions": [
+          "search"
+      ],
+      "indexes": [
+          "doctors"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    },
+    {
+      "description": "Default Admin API Key (Use it for all other operations. Caution! Do not share it on the client side)",
+      "key": "380689dd379232519a54d15935750cc7625620a2ea2fc06907cb40ba5b421b6f",
+      "actions": [
+          "*"
+      ],
+      "indexes": [
+          "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    },
+    {
+      "description": "Search patient records key",
+      "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+      "actions": [
+        "search"
+      ],
+      "indexes": [
+        "patient_medical_records"
+      ],
+      "expiresAt": "2023-01-01T00:00:00Z",
+      "createdAt": "2022-01-01T10:00:00Z",
+      "updatedAt": "2022-01-01T10:00:00Z"
+    }
+  ]
+}
+```
+
+### Deleting an API key
+
+If a key is no longer useful or has been compromised, you can use [delete key endpoint](/reference/api/keys.md#delete-a-key) to disable it before its expiry date.
+
+If we accidentally exposed our `Search patient records key`, we can delete it to prevent unauthorized parties from gaining access to our `patient_medical_records` index:
+
+<CodeSamples id="security_guide_delete_key_1" />
+
+### Expired keys
+
+Once a key is past its `expiresAt` date, using it when communicating with a protected instance will return an error.
```

Comment:
```suggestion
Once a key is past its `expiresAt` date, using it for API authorization will return an error. It will still be returned by the [list keys endpoint](/reference/api/keys.md#get-all-keys).
```